### PR TITLE
Complete Buildpack

### PR DESCRIPTION
**Pull request by jedkirby**
This is just the core code. I still need to do the following, which are issues already:

- Write Documentation - https://github.com/jedkirby/heroku-buildpack-typekit/issues/2
- Provide Full Test Suite - https://github.com/jedkirby/heroku-buildpack-typekit/issues/3

### CODE REVIEW

File: tests/envs/TYPEKIT_API_KEY [hunk 1]
```diff
@@ -0,0 +1 @@
+78ybi67812345niy8902345n9679123
```
Comment (steadweb) [inline tests/envs/TYPEKIT_API_KEY:1]: Assuming this is ok to be published?

File: tests/envs/TYPEKIT_KIT_ID [hunk 2]
```diff
@@ -0,0 +1 @@
+mkjg5fub
```
Comment (steadweb) [inline tests/envs/TYPEKIT_KIT_ID:1]: Assuming this is ok to be published?

File: tests/envs/HEROKU_APP_NAME [hunk 3]
```diff
@@ -0,0 +1 @@
+website-develop
```
Comment (steadweb) [inline tests/envs/HEROKU_APP_NAME:1]: Will this ever change?

File: bin/compile [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    domains = kit['kit']['domains'];
+
+    if herokuAppUrl in domains:
+        print output.line('The domain "' + herokuAppUrl + '" has already been loaded into Typekit.');
+    else:
+        domains.append(herokuAppUrl);
+
+        if not typekit.add(domains):
+            raise RuntimeError('Error');
```
Comment (jedkirby) [inline bin/compile]: This should not be just `Error`, but be more descriptive.

File: bin/compile [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+        typekit.publish();
+
+        print output.line('The domain "' + herokuAppUrl + '" has been added to your Typekit account.');
+        print output.line('Your new kit may take a few minutes to be completely distributed across the Typekie network.');
+
+except RuntimeError as e:
```
Comment (jedkirby) [inline bin/compile:77]: Possibly output the response of `e` within the display message(s)..

**Review by steadweb — APPROVED** [review]
Logic is sound. Nothing major in terms of style as it's a script. Couple of env questions which you might deem not an issue. Otherwise :+1:

File: bin/release [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env python
+
+##
+# NB: This has not yet been implements.
```
Comment (jedkirby) [inline bin/release]: Spelling mistake, should be "implemented".

File: tests/envs/TYPEKIT_API_KEY (hunk 1 above)
Comment (jedkirby) [reply tests/envs/TYPEKIT_API_KEY:1]: Yep, this is all good - it's a made up string, will be used for testing eventually.

File: tests/envs/TYPEKIT_KIT_ID (hunk 2 above)
Comment (jedkirby) [reply tests/envs/TYPEKIT_KIT_ID:1]: This is also the same as the `TYPEKIT_API_KEY` - all good also.

File: tests/envs/HEROKU_APP_NAME (hunk 3 above)
Comment (jedkirby) [reply tests/envs/HEROKU_APP_NAME:1]: Yep, they'll change depending on what Heroku sends, this one's just for testing though.

File: bin/compile [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    else:
+        domains.append(herokuAppUrl);
+
+        if not typekit.add(domains):
+            raise RuntimeError('Error');
+
+        typekit.publish();
+
+        print output.line('The domain "' + herokuAppUrl + '" has been added to your Typekit account.');
+        print output.line('Your new kit may take a few minutes to be completely distributed across the Typekie network.');
```
Comment (steadweb) [inline bin/compile:75]: Spelling mistake:

`Typekie` should be `Typekit`

**Pull request merged**